docs/src/main/paradox/java/http/handling-blocking-operations-in-akka-http-routes.md

```diff
@@ -0,0 +1,120 @@
+lead to problems. It is important to handle the blocking operations correctly.
+
+## Problem
+
+Using `context.dispatcher()` as the dispatcher on which the blocking Future
+executes can be a problem - the same dispatcher is used by the routing
+infrastructure to actually handle the incoming requests. 
+
+If all of the available threads are blocked, the routing infrastructure will end up *starving*. 
+Therefore, routing infrastructure should not be blocked. Instead, a dedicated dispatcher
+for blocking operations should be used.
+
+@@@ note
+
+Blocking APIs should also be avoided if possible. Try to find or build Reactive APIs,
+such that blocking is minimised, or moved over to dedicated dispatchers.
+
+Often when integrating with existing libraries or systems it is not possible to
+avoid blocking APIs. The following solution explains how to handle blocking
+operations properly.
+
+Note that the same hints apply to managing blocking operations anywhere in Akka,
+including in Actors etc.
+
+@@@
+
+In the thread state diagrams below the colours have the following meaning:
+
+ * Turquoise - Sleeping state
+ * Orange - Waiting state
+ * Green - Runnable state
+
+The thread information was recorded using the YourKit profiler, however any good JVM profiler 
+has this feature (including the free and bundled with the Oracle JDK VisualVM, as well as Oracle Flight Recorder). 
+
+### Problem example: blocking the default dispatcher
+
+@@snip [BlockingInHttpExamples.java](../../../../test/java/docs/http/javadsl/server/BlockingInHttpExamples.java) { #blocking-example-in-default-dispatcher }
+
+Here the app is exposed to a load of continuous GET requests and large numbers
+of akka.actor.default-dispatcher threads are handling requests. The orange
+portion of the thread shows that it is idle. Idle threads are fine -
+they're ready to accept new work. However, large amounts of Turquoise (sleeping) threads are very bad!
+
+![DispatcherBehaviourOnBadCode.png](DispatcherBehaviourOnBadCode.png)
+
+After some time, the app is exposed to the load of POST requests,
+which will block these threads. For example "`default-akka.default-dispatcher2,3,4`"
+are going into the blocking state, after having been idle. It can be observed
+that the number of new threads increases, "`default-akka.actor.default-dispatcher 18,19,20,...`" 
+however they go to sleep state immediately, thus wasting the
+resources.
+
+The number of such new threads depends on the default dispatcher configuration,
+but it will likely not exceed 50. Since many POST requests are being processed, the entire
+thread pool is starved. The blocking operations dominate such that the routing
+infra has no thread available to handle the other requests.
+
+In essence, the `Thread.sleep()` operation has dominated all threads and caused anything 
+executing on the default dispatcher to starve for resources (including any Actors
+that you have not configured an explicit dispatcher for (sic!)).
```

Comment:
I don't know either, I copied the text from the Scala version. If it's there maybe it's because somebody thought it was "needed?